pkg/build/rpm.go

```diff
@@ -0,0 +1,55 @@
+package build
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+func (b *Builder) getRPMFileNames() error {
+	b.rpmSourceDir = filepath.Join(b.buildConfig.ImageConfigDir, "rpms")
```

Comment:
This is another state modifier which shouldn't be included. In general, we don't store the results of one-off tasks e.g. we do need to store `combustionScripts` because a lot of components will use that variable to store data but the builder itself does not care which directory is being used for RPMs.

Ideally, the RPM source directory should be a local variable which we can pass wherever necessary.

Reply:
👍  will update